src/Tribe/JSON_LD/Ticket.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+// Don't load directly
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+/**
+ * A JSON-LD class extended of the Abstract that lies on the Common Package
+ * Used for generating a Event JSON-LD markup
+ */
+class Tribe__Events__JSON_LD__Event extends Tribe__JSON_LD__Abstract {
+
+	/**
+	 * Which type of element this actually is
+	 *
+	 * @see https://developers.google.com/structured-data/rich-snippets/
+	 * @var string
+	 */
+	public $type = 'Offer';
+
+	/**
+	 * On PHP 5.2 the child class doesn't get spawned on the Parent one, so we don't have
```

Comment:
I could be misunderstanding ... but if we used `get_class( $this )` (in the superclass) mightn't that clean up the do-not-use-param in the method signatures of the subclasses in a 5.2-compatible way?